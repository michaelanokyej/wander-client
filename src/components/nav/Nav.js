import React from "react";
import "./NavBar.css";
import { Link } from "react-router-dom";
import DrawerToggleButton from "../sidedrawer/DrawerToggleButton";
import toursAndUserContext from "../context/toursAndUserContext.js";

class NavBar extends React.Component {
  static contextType = toursAndUserContext;

  // state = {
  //   disabled: ""
  // }

  // disableLink = () => {
  //   if(!this.context.loggedIn) {
  //     this.setState({
  //       disabled: "disabled"
  //     });
  //   }
  // }

  render() {
    return (
      <header className="toolbar">
        <nav className="toolbar__navigation">
          {/* <div className="toolbar__toggle-button">
            <DrawerToggleButton click={this.props.drawerClickHandler}/>
          </div> */}
          <div className="toolbar__logo">
            <Link to="/">
              <img src="IMG_2340.png" alt="logo" />
            </Link>
          </div>
          <div className="spacer" />
          <div className="toolbar_navigation_items">
            <ul>
              {this.context.loggedIn ? (
                <>
                  <li>
                    <Link to="/home">Home</Link>
                  </li>
                  <li>
                    <Link to="/post-a-tour">Post Tour</Link>
                  </li>

                  <li>
                    {" "}
                    <Link to="/users/:username">
                      {this.context.userName || "Profile"}
                    </Link>
                  </li>
                </>
              ) : (
                <li>
                  <Link to="/signup">Sign Up</Link>
                </li>
              )}
            </ul>
          </div>
          <div className="toolbar__toggle-button">
            <DrawerToggleButton click={this.props.drawerClickHandler} />
          </div>
        </nav>
      </header>
    );
  }
}

export default NavBar;
