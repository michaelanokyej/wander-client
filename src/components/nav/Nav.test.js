import React from "react";
import Nav from "./Nav"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"

describe(`Nav component`, () => {
    it(`renders the Nav component`, () => {
        const wrapper = shallow(<Nav />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});