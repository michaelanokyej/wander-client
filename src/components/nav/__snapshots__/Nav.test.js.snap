// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders the Nav component 1`] = `
<header
  className="toolbar"
>
  <nav
    className="toolbar__navigation"
  >
    <div
      className="toolbar__logo"
    >
      <Link
        to="/"
      >
        <img
          alt="logo"
          src="IMG_2340.png"
        />
      </Link>
    </div>
    <div
      className="spacer"
    />
    <div
      className="toolbar_navigation_items"
    >
      <ul>
        <li>
          <Link
            to="/signup"
          >
            Sign Up
          </Link>
        </li>
      </ul>
    </div>
    <div
      className="toolbar__toggle-button"
    >
      <DrawerToggleButton />
    </div>
  </nav>
</header>
`;
